src/config_validator.py

```
@@ -0,0 +1,156 @@
+            float: Validated timeout value
+        
+        Raises:
+            ConfigValidationError: If timeout is invalid
+        """
+        try:
+            timeout_float = float(timeout)
+            if timeout_float <= 0 or timeout_float > 120:  # Reasonable timeout range
+                raise ConfigValidationError(f"Invalid timeout value: {timeout}")
+            return timeout_float
+        except (TypeError, ValueError):
+            raise ConfigValidationError(f"Invalid timeout value: must be a numeric value, got {type(timeout)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)